test/ParserTests.hs

```diff
@@ -68,6 +70,17 @@ testCheckLeft2 = TestCase (
   True
   (isLeft $ parseAll xtype "" "(22222)"))
 
+testFunctionType1 :: Test
+testFunctionType1 = TestCase (
+  assertEqual "Function non tuple types"
+  (Right (Ft intxt intxt))
+  (parseAll ftype "" "Int -> Int"))
+
+testFunctionType2 :: Test
+testFunctionType2 = TestCase (
+  assertEqual "Function tuple types"
+  (Right (Ft (Tup [Tup [intxt, intxt], boolxt]) (Tup [intxt, intxt])))
+  (parseAll ftype "" "((Int, Int), Bool) -> (Int, Int)"))
```

Comment:
Passing cases look good, but could you also add in:
- `(Int,Int) -> Int`
- `Int -> (Int,Int)` (just to be sure that both sides will work properly)
- and maybe 1 or 2 failing cases...I can't think of any edges currently beyond the two I just listed...but perhaps we may already have test cases elsewhere that already cover things nicely.